internal/controller/pgcluster/pgclustercontroller.go

```diff
@@ -348,11 +345,58 @@ func (c *Controller) onUpdate(oldObj, newObj interface{}) {
 
 // onDelete is called when a pgcluster is deleted
 func (c *Controller) onDelete(obj interface{}) {
-	// cluster := obj.(*crv1.Pgcluster)
-	//	log.Debugf("[Controller] ns=%s onDelete %s", cluster.ObjectMeta.Namespace, cluster.ObjectMeta.SelfLink)
+	ctx := context.TODO()
+	cluster := obj.(*crv1.Pgcluster)
+
+	log.Debugf("pgcluster onDelete for cluster %s (namespace %s)", cluster.Name, cluster.Namespace)
+
+	// a quick guard: see if the "rmdata Job" is running.
+	options := metav1.ListOptions{
+		LabelSelector: fields.AndSelectors(
+			fields.OneTermEqualSelector(config.LABEL_PG_CLUSTER, cluster.Name),
+			fields.OneTermEqualSelector(config.LABEL_RMDATA, config.LABEL_TRUE),
+		).String(),
+	}
+
+	jobs, err := c.Client.BatchV1().Jobs(cluster.Namespace).List(ctx, options)
+
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	// iterate through the list of Jobs and see if any are currently active
+	// a succeeded Job could be a remnaint of an old Job for the cluser of a
+	// same name, in which case, we can continue with deleting the cluster
+	for _, job := range jobs.Items {
+		// if the Job is currently active or if the Job is not active but never has
+		// completed, exit
+		if job.Status.Active > 0 {
+			return
+		} else if job.Status.Succeeded < 1 {
+			return
```

Comment:
this part here seems confusing, took me a few mins to understand it
if the job is active exit, if the job is not active but never succeeded exit...do we log that somewhere that we can't delete b/c of an active job or a job never succeeded?

otherwise lgtm

Reply:
@ValClarkson I clarified the comment. However, looking through it, I realized that the "failure" cause was not great, so I added this additional logic. Please let me know what you think.

Reply:
I agree this is much better with the backofflimit added